compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -1543,7 +1543,9 @@ fn assemble_candidates_from_trait_def<'cx, 'tcx>(
     // Check whether the self-type is itself a projection.
     // If so, extract what we know from the trait and try to come up with a good answer.
     let bounds = match *obligation.predicate.self_ty().kind() {
-        ty::Alias(_, ref data) => tcx.item_bounds(data.def_id).subst(tcx, data.substs),
+        ty::Alias(ty::Projection | ty::Opaque, ref data) => {
```

Comment:
Maybe worth a comment saying that we don't have item bounds for IAT?

Reply:
Yeah, can do